ml-agents/mlagents/trainers/bc/trainer.py

```diff
@@ -9,7 +9,7 @@
 from mlagents.envs.brain import BrainInfo
 from mlagents.envs.action_info import ActionInfoOutputs
 from mlagents.trainers.bc.policy import BCPolicy
-from mlagents.trainers.buffer import Buffer
+from mlagents.trainers.buffer import AgentBuffer, AgentProcessorBuffer
```

Comment:
Maybe the names are not descriptive enough about the difference between the two.

Reply:
What about calling it the `ProcessingBuffer`? 

In the next PR I'm actually removing the AgentProcessorBuffer entirely and baking the logic into the AgentProcessor itself.

Reply:
Processing is a little vague. I would say maybe SingleAgentExperienceBuffer, MultiAgentExperienceBuffer ?